### PR TITLE
DOC cosmetic changes to `prevalence_correction.py`

### PR DESCRIPTION
**Pull request by AnneBeyer**
I mainly moved the model comparator definition to the front and renamed `true_positive_rate_past` to `true_prevalence_past`, because my brain kept activating `TPR` whenever I read it :sweat_smile:, and changed the order in one Exercise set to untangle the answers a bit. I think this is good to go for tomorrow.

I also checked with Claude if things have changed in the meantime, and it pointed me to the following two publications, confirming that this is still one of the most valid approaches (assuming that the model is well-calibrated on the training set):

Saerens et al. (2002)[^1] independently proposed a similar solution roughly at the same time, using an EM algorithm to jointly compute the probabilities **and** estimate the underlying a priori class distribution from an unlabeled dataset. Their E-step is essentially the same formula as Theorem 2 proposed by Elkan et al. for a given new target distribution `b'`.

Alexandari et al. (2020)[^2] more recently confirmed that, if the underlying model is well calibrated on the training data, this EM algorithm is still one of the best available options to address label shift.

The approach by Saerens et al., together with the heuristics proposed by Alexandari et al. (and many alternatives that have also been proposed over the last years), are implemented in the (scikit-learn compatible and actively maintained) package [QuaPy](https://hlt-isti.github.io/QuaPy/quapy.method.html#quapy.method.aggregative.EMQ). I haven't had time to play around with this package yet, though, so I didn't include it in the tutorial.

[^1]: [Saerens, Latinne & Decaestecker, "Adjusting the Outputs of a Classifier to New A Priori Probabilities: A Simple Procedure", Neural Computation 14(1):21–41, 2002](https://dl.acm.org/doi/10.1162/089976602753284446) 
[^2]: [Alexandari, Kundaje & Shrikumar, "Maximum Likelihood with Bias-Corrected Calibration is Hard-to-Beat at Label Shift Adaptation," ICML 2020](https://proceedings.mlr.press/v119/alexandari20a.html)

### CODE REVIEW

**Review by glemaitre — APPROVED** [review]

**Comment by glemaitre** [comment]
LGTM. @AnneBeyer feel free to merge.
